src/components/PopupWithForm.js

```diff
@@ -24,11 +25,16 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", this._onPopupSubmit);
   }
 
-  _onPopupSubmit = (evt) => {
-    const event = evt;
-    this._handleFormSubmit(this._getInputValues(), event);
+  closeAfterSubmit() {
     this._popupForm.reset();
+    this._submitButton.textContent = this._submitButtonText;
     this._formValidator.toggleButtonState();
     this.close();
+  }
+
+  _onPopupSubmit = (evt) => {
+    const event = evt;
+    this._submitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```